halfpipe/ingest/tests/test_resolve.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+
+import json
+import time
+from datetime import datetime
+from pathlib import Path
+from typing import List
+
+import pytest
+import requests
+
+from ...model.file.base import File
+from ...model.spec import Spec
+from ..resolve import ResolvedSpec
+
+
+@pytest.mark.parametrize(
+    ("openneuroID"),
+    (
+        ("ds004109"),
+        ("ds004187"),
+        ("ds004161"),
+    ),
+)
+def test__resolve_bids(tmp_path: Path, openneuroID: str):
+    # Get file names with GraphQL request
+    # Test different IDs with .parametrize or string.Template substitute method
+    query_example = f"""query{{
+        dataset (id: "{openneuroID}"){{
+            latestSnapshot{{
+                id
+                files(prefix: null){{
+                    filename
+                }}
+            }}
+        }}
+    }}"""
+    gql_url = "https://openneuro.org/crn/graphql"
+    attempts = 0
+    while attempts < 3:
+        r = requests.post(gql_url, json={"query": query_example})
+        if not r == 200:
+            time.sleep(0.3)
+            attempts += 1
+            continue
+        break
+
+    # r = requests.post(gql_url, json={"query": query_example})
+    # if not r == 200:
+    #   raise RuntimeError("Could not fetch file listing")
```

Comment:
```suggestion
    if not r == 200:
      raise RuntimeError("Could not fetch file listing")
```

I think it would be good to keep this to know if the retries worked ;-)

Reply:
Totally, I forgot I commented the line and not put it back in..